js/foam/u2/ScrollView.js

```diff
@@ -0,0 +1,617 @@
+    </ul>
+  */},
+
+  constants: {
+    ROW_CLICK: ['row-click'],
+  },
+
+  properties: [
+    {
+      type: 'foam.core.types.DAO',
+      name: 'data',
+      onDAOUpdate: 'onDAOUpdate',
+    },
+    {
+      name: 'model',
+      documentation: 'The model for the data. Defaults to the DAO\'s model.',
+      defaultValueFn: function() { return this.data.model; }
+    },
+    {
+      name: 'runway',
+      defaultValue: 800,
+      documentation: 'The distance in pixels to render on either side of the viewport. Defaults to 800.'
+    },
+    {
+      name: 'count',
+      defaultValue: 0,
+      postSet: function(old, nu) {
+        this.scrollHeight = nu * this.rowHeight;
+      }
+    },
+    {
+      name: 'scrollHeight',
+      documentation: 'The total height of the scrollable pane. Generally <tt>count * rowHeight</tt>.',
+    },
+    {
+      name: 'scrollTop',
+      documentation: 'Set on each scroll event. Saved so we can resume a previous scroll position when we return to this view from elsewhere.',
+      defaultValue: 0
+    },
+    {
+      name: 'scrollGesture',
+      hidden: true,
+      transient: true,
+      lazyFactory: function() {
+        return this.GestureTarget.create({
+          containerID: this.$scroller.id,
+          handler: this,
+          gesture: 'verticalScrollNative'
+        });
+      }
+    },
+    {
+      name: 'rowHeight',
+      defaultValue: -1,
+      documentation: function() {/*
+        <p>The height of each row in CSS pixels.</p>
+        <p>If specified, <tt>ScrollView</tt> will use this height. Otherwise it will use <tt>rowView</tt>'s <tt>preferredHeight</tt>, if set. Otherwise defaults to <tt>-1</tt> until it can be computed dynamically.</p>
+
+        <p>That computation requires rendering a <tt>rowView</tt> without its <tt>data</tt> set, which breaks some views. If that happens to you, or the size of an empty <tt>rowView</tt> is smaller than a full one, measure the proper height and set <tt>preferredHeight</tt> on the <tt>rowView</tt>, or <tt>rowHeight</tt> on the <tt>ScrollView</tt>.</p>
+      */}
+    },
+    {
+      name: '$rowSize',
+      hidden: true,
+      documentation: 'If the <tt>rowHeight</tt> is not set, a <tt>rowView</tt> will be constructed and its height checked. This property holds that view so it can be destroyed properly.'
+    },
+    {
+      model_: 'ViewFactoryProperty',
+      name: 'rowView',
+      documentation: 'The view for each row. Can specify a <tt>preferredHeight</tt>, which will become the <tt>rowHeight</tt> for the <tt>ScrollView</tt> if <tt>rowHeight</tt> is not set explicitly.',
+      postSet: function(_, nu) {
+        if (typeof nu === 'string' && this.rowHeight < 0) {
+          var view = this.Y.lookup(nu);
+          if ( view.PREFERRED_HEIGHT )
+            this.rowHeight = view.create({ model: this.data.model }).preferredHeight;
+        }
+      }
+    },
+    {
+      name: '$container',
```

Comment:
This naming made sense in u1 when we accessed elements with .$, but in U2 this doesn't exist, so a better convention would be to name this containerE, scrollerE, etc.